### PR TITLE
feat: dto work for get positions

### PR DESCRIPTION
**Pull request by KaloDFreelancer**

### CODE REVIEW

File: src/types/offers.ts [hunk 1]
```diff
@@ -53,12 +53,12 @@ export const OfferSchema = UniqueOfferSchema.and(
...
 export type Offer = z.infer<typeof OfferSchema>;
 
-enum ProviderType {
+export enum ProviderType {
   ASTARIA = 'ASTARIA',
   BLUR = 'BLUR',
   X2Y2 = 'X2Y2',
 }
-enum LoanType {
+export enum LoanType {
```
Comment (mlenser) [inline src/types/offers.ts:61]: We should remove these exports unless they are needed by someone external (us on BE, us on FE, or some 3rd party partner).

File: src/types/position.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  asset: AssetSchema,
+  loan: LoanPositionSchema,
+});
+export type Position = z.infer<typeof PositionSchema>;
+
+export const PositionsResponseSchema = z.object({
+  paging: PagingSchema,
+  positions: PositionSchema.array(),
+});
+export type PositionsResponse = z.infer<typeof PositionsResponseSchema>;
```
Comment (mlenser) [inline src/types/position.ts:22]: The naming could be more understandable imo.

We should rename `LoanSchema` to `LoanBaseSchema` and use that for `AssetOffersSchema` as well as below:

```suggestion
export const LoanSchema = LoanBaseSchema.extend({
  created: z.number(),
  id: z.string(),
});

export const LoanPositionSchema = z.object({
  asset: AssetSchema,
  loan: LoanSchema,
});
export type LoanPosition = z.infer<typeof LoanPositionSchema>;

export const LoanPositionsResponseSchema = z.object({
  paging: PagingSchema,
  positions: LoanPositionSchema.array(),
});
export type LoanPositionsResponse = z.infer<typeof LoanPositionsResponseSchema>;
```

File: src/types/offers.ts (hunk 1 above)
Comment (KaloDFreelancer) [reply src/types/offers.ts:61]: They are imported with `LoanBaseSchema` in `position.ts` and the build fails if we don't export them.

Comment (mlenser) [reply src/types/offers.ts:61]: image
They aren't imported in `positions.ts`

Comment (KaloDFreelancer) [reply src/types/offers.ts:61]: This is what I get when I remove the exports. We need them for the `LoanBaseSchema` import.
```
Exported variable 'LoanSchema' has or is using name 'LoanType' from external module "/Users/kaloyanmitev/Astaria/astaria-sdk/src/types/offers" but cannot be named.
```

Comment (mlenser) [reply src/types/offers.ts:61]: Interesting! I tried it locally and confirmed. I didn't know Zod handled it like this.

**Review by mlenser — APPROVED** [review]

**Pull request merged**